OpenRA.Mods.Common/Util.cs

```diff
@@ -228,5 +230,22 @@ public static string FriendlyTypeName(Type t)
 
 			return t.Name;
 		}
+
+		public static int GetProjectileInaccuracy(int baseInaccuracy, InaccuracyType inaccuracyType, ProjectileArgs args)
+		{
+			var inaccuracy = ApplyPercentageModifiers(baseInaccuracy, args.InaccuracyModifiers);
+			switch (inaccuracyType)
+			{
+				case InaccuracyType.Maximum:
+					var weaponMaxRange = ApplyPercentageModifiers(args.Weapon.Range.Length, args.RangeModifiers);
+					return inaccuracy * (args.PassiveTarget - args.Source).Length / weaponMaxRange;
+				case InaccuracyType.PerCellIncrement:
+					return inaccuracy * (args.PassiveTarget - args.Source).Length / 1024;
+				case InaccuracyType.Absolute:
+					return inaccuracy;
+				default:
+					return 0;
```

Comment:
This case should never be hit, right? I assume this is then to silence some sort of compiler warning. Wouldn't it make more sense to throw an exception instead?

Reply:
IMO it would be reasonable to default unknown `InaccuracyType`s to `InaccuracyType.Absolute`, so we could just add the `default` label to that case?

Reply:
Having the default of the TraitInfo field be `Maximum` but then defaulting this to `Absolute` would be weird.

Reply:
@abcdefg30 can you confirm that this resolves your question or suggest a suitable fix so we can move this forward?

Reply:
I thought the result here was defaulting to `Maximum` since that is the default on the info field?

Reply:
As you initially assumed, this is only here to please the compiler, because otherwise we get an error (not a warning) - `not all code paths return a value`. So one way or another we need a `default` case or a `return` after the `switch` (or to throw an exception).
`InaccuracyType.Absolute` might be the most logical default, but as I pointed out, it needs to be the default in both places and one of them would require a major update rule that really isn't needed.
The `default` here results in no inaccuracy, actually. So if one sets up their `InaccuracyType` incorrectly they will get no inaccuracy ingame. Hopefully that would be enough to make them realize the error without having to throw an exception.

Reply:
Can we compromise with a `throw new InvalidEnumArgumentException("inaccuracyType", (int)inaccuracyType, typeof(InaccuracyType));` so we can merge this and move on to more productive arguments?

Reply:
I have nothing against throwing an exception if people are fine with that.